src/Workspaces/Core/Portable/Workspace/Solution/ProjectState.cs

```diff
@@ -325,8 +338,10 @@ public TextDocumentState GetAdditionalDocumentState(DocumentId documentId)
             ImmutableDictionary<DocumentId, DocumentState> documentStates = null,
             ImmutableDictionary<DocumentId, TextDocumentState> additionalDocumentStates = null,
             AsyncLazy<VersionStamp> latestDocumentVersion = null,
-            AsyncLazy<VersionStamp> latestDocumentTopLevelChangeVersion = null)
+            AsyncLazy<VersionStamp> latestDocumentTopLevelChangeVersion = null,
+            ValueSource<ProjectStateChecksums> lazyChecksums = null)
         {
+            // for now, re-calculate all checksums when anything changed
```

Comment:
this doesn't seem to be the case.  By default you pass along _lazyChecksums if lazyChecksums is not changed.  That means you're using the old checksum values when anything changes.